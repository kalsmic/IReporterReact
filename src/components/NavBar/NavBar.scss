.menu {
  &__list {
    list-style: none;
    margin: 0;
    padding: 0;
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    flex-wrap: wrap;

    &__item {
      color: #37C0E6;
      text-decoration: none;

      &__active {
        color: #ADFF2F;
        text-decoration: underline;
      }
    }
  }
}

@media (min-width: 700px) {
  .menu {
    &__list {
      flex-direction: column;
    }
  }
}
