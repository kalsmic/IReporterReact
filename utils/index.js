export const findByTestAttr = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const baseURL = 'https://ireporterv3.herokuapp.com/api/v3';

export const isAuthenticated = () => {
  const token = sessionStorage.getItem('iReporterToken');
  if (token) { return true; }
  return false;
};
