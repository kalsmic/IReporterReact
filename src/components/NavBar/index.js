import React from 'react';
import { NavLink } from 'react-router-dom';
import './NavBar.scss';

const NavBar = () => (
  <div className="menu">
    <ul className="menu__list ">
      <li>
        <NavLink
          to="/createIncident"
          exact
          activeClassName="menu__list__item__active"
          className="menu__list__item"
        >
          <i className="fa fa-plus" />
          {' '}
          Create Incident
        </NavLink>
      </li>
      <li>
        <NavLink
          to="/red-flags"
          exact
          activeClassName="menu__list__item__active"
          className="menu__list__item"
        >
          <i className="fa fa-flag" />
          {' '}
Redflags
        </NavLink>
      </li>
      <li>
        <NavLink
          to="/interventions"
          exact
          activeClassName="menu__list__item__active"
          className="menu__list__item"
        >
          <i className="fa fa-drafting-compass" />
          {' '}
Interventions
        </NavLink>
      </li>

      <li>
        <NavLink
          to="/"
          data-test="logout"
          exact
          activeClassName="menu__list__item__active"
          className="menu__list__item"
          onClick={() => {
            sessionStorage.removeItem('iReporterToken');
          }}
        >
          <i className="fa fa-unlock" />
          {' '}
logout
        </NavLink>
      </li>
    </ul>
  </div>
);

export default NavBar;
