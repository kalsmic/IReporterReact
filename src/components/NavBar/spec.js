import React from 'react';
import { shallow } from 'enzyme';
import NavBar from './index';
import { findByTestAttr } from '../../../utils';


describe('NavBar', () => {
  const component = shallow(<NavBar />);

  it('should render without crashing', () => {
    expect(component).toMatchSnapshot();
  });

  it('should clear the user session on logout', () => {
    const logoutLink = findByTestAttr(component, 'logout');
    const logoutfn = jest.fn();
    logoutLink.simulate('click');
    console.log(component.debug());
  });
});
