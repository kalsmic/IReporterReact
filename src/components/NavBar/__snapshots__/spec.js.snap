// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar should render without crashing 1`] = `
<div
  className="menu"
>
  <ul
    className="menu__list "
  >
    <li>
      <NavLink
        activeClassName="menu__list__item__active"
        aria-current="page"
        className="menu__list__item"
        exact={true}
        to="/createIncident"
      >
        <i
          className="fa fa-plus"
        />
         
        Create Incident
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="menu__list__item__active"
        aria-current="page"
        className="menu__list__item"
        exact={true}
        to="/red-flags"
      >
        <i
          className="fa fa-flag"
        />
         
        Redflags
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="menu__list__item__active"
        aria-current="page"
        className="menu__list__item"
        exact={true}
        to="/interventions"
      >
        <i
          className="fa fa-drafting-compass"
        />
         
        Interventions
      </NavLink>
    </li>
    <li>
      <NavLink
        activeClassName="menu__list__item__active"
        aria-current="page"
        className="menu__list__item"
        data-test="logout"
        exact={true}
        onClick={[Function]}
        to="/"
      >
        <i
          className="fa fa-unlock"
        />
         
        logout
      </NavLink>
    </li>
  </ul>
</div>
`;
